Update build/deploy instructions (#3309)
wpilibsuite/frc-docs: source/docs/software/vscode-overview/deploying-robot-code.rst

Robot projects must be compiled ("built") and deployed in order to run on Systemcore.  Since the code is not compiled natively on the robot controller, this is known as "cross-compilation."

## Build

To build a robot project, do one of:

> 1. In Visual Studio Code, click the WPILib logo in the top right to launch the WPILib Command Palette (or type :kbd:`ctrl+shift+p` and then type WPILib) and enter/select :guilabel:`Build Robot Code`

Reviewer: I don't think it's accurate to "type" a keyboard shortcut since that generally implies that the user is pressing one key at a time. Maybe just say "use"?

Reviewer 2: I think "press" is probably clearer.
```suggestion
1. In Visual Studio Code, click the WPILib logo in the top right to launch the WPILib Command Palette (or press :kbd:`ctrl+shift+p` and then type WPILib) and enter/select :guilabel:`Build Robot Code`
```

2. Open the shortcut menu indicated by the ellipses in the top right corner of the VS Code window and select :guilabel:`Build Robot Code`

> 3. Right-click on the build.gradle file in the project hierarchy and select :guilabel:`Build Robot Code`

Reviewer 2: Now that I'm looking at this again, the numbered list kinda gives the impression it's a list of steps to follow in order. This probably makes more sense as a bulleted list?

I know it was like that before though, so maybe this came up before and I'm missing the context there.

If successful, we will see a "Build Successful" message (1) and the RioLog will open with the console output from the robot program as it runs (2).

.. image:: images/deploying-robot-code/build-successful.png

> .. note:: The run button in VS Code's debug view is not used to run robot code. Instead, use the :guilabel:`Deploy Robot Code` command as described above. The debug view's run button is used for running and debugging code on the local machine in simulation, which is not applicable for robot code that runs on Systemcore.

Reviewer 2: The end of this sentence suggests that robot code that is able to run in simulation won't run on Systemcore, which is of course not necessarily true.

```suggestion
.. note:: The run button in VS Code's debug view is not used to run robot code. Instead, use the :guilabel:`Deploy Robot Code` command as described above. The debug view's run button is used for running and debugging code on the local machine in simulation, which will not run the robot code on a Systemcore.
```

## Build vs Deploy

Choosing to deploy robot code will first build the code if it has not been built since the last change. If the code has already been built, it will skip the build step and deploy the existing build artifacts to Systemcore. You can force a rebuild by selecting :guilabel:`Build Robot Code` before deploying. There are some differences when running the :guilabel:`Build Robot Code` command vs the :guilabel:`Deploy Robot Code` command:

> 1. :guilabel:`Build Robot Code` will download dependencies from the internet, while :guilabel:`Deploy Robot Code` will not. If you have not built the code before (for example, if you are working with a new project or you have recently cloned the repository, or you have added or changed a vendordep), you must run :guilabel:`Build Robot Code` at least once to download dependencies before you can deploy.

Reviewer 3: gradle deploy also triggers compilation (it depends on compile tasks), so Gradle will still attempt dependency resolution (which will try downloads) when you deploy.  I think we should remove or rephrase.

Reviewer 3: ```suggestion
1. Running :guilabel:`Build Robot Code` before going to a competition ensures all dependencies are downloaded and cached locally. This is important because the build process requires an internet connection to download dependencies the first time, and competition venues may not have reliable internet access.
```

Author: > gradle deploy also triggers compilation (it depends on compile tasks), so Gradle will still attempt dependency resolution (which will try downloads) when you deploy

deploy does depend on compile, but deploy specifically uses the --offline flag, so it will still attempt dependency resolution for any offline repos, but it will not try online dependency resolution, so I think the way it's worded is still accurate.

2. :guilabel:`Build Robot Code` will run unit tests, if your project has unit tests, while :guilabel:`Deploy Robot Code` will not.

## Clean

> If you want to force a full rebuild of the code, you can select :guilabel:`Run Gradle Clean` the WPILib Menu. This will delete all build artifacts and force a full rebuild the next time you build or deploy.

Reviewer 3: ```suggestion
If you want to force a full rebuild of the code, you can select :guilabel:`Run Gradle Clean` from the WPILib Menu. This will delete all build artifacts and force a full rebuild the next time you build or deploy.
```
